subprojects/core/src/test/groovy/org/gradle/internal/vfs/DarwinFileWatcherRegistryTest.groovy

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.vfs
+
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.nio.file.Paths
+
+@Requires(TestPrecondition.MAC_OS_X)
```

Comment:
How about using `NOT_WINDOWS`? The test should work on Linux as well and that would give earlier feedback.

Reply:
This is a watcher explicitly designed and only used on macOS, why should we test it on any other platform?

Reply:
Since the Unit test should actually run anywhere. For integration tests, I fully agree that it should run only on macOS.

Reply:
With `NOT_WINDOWS` the unit test still wouldn't "run anywhere." For a piece of code that is specifically for a give OS I don't see this as an improvement.